rpcserverhandlers_test.go

```diff
@@ -1975,6 +1994,122 @@ func TestHandleEstimateSmartFee(t *testing.T) {
+			chain.estimateNextStakeDifficultyQueue = validQueueFunc()
+			return chain
+		}(),
+		result: &types.EstimateStakeDiffResult{
+			Min:      dcrutil.Amount(low).ToCoin(),
+			Max:      dcrutil.Amount(high).ToCoin(),
+			Expected: dcrutil.Amount(med).ToCoin(),
+			User:     dcrjson.Float64(dcrutil.Amount(user).ToCoin()),
+		},
+	}, {
+		name:    "handleEstimateStakeDiff: ok no Tickets arg",
+		handler: handleEstimateStakeDiff,
+		cmd:     &types.EstimateStakeDiffCmd{},
+		mockChain: func() *testRPCChain {
+			chain := defaultMockRPCChain()
+			chain.estimateNextStakeDifficultyQueue = validQueueFunc()
+			return chain
+		}(),
+		result: &types.EstimateStakeDiffResult{
+			Min:      dcrutil.Amount(low).ToCoin(),
+			Max:      dcrutil.Amount(high).ToCoin(),
+			Expected: dcrutil.Amount(med).ToCoin(),
+		},
+	}, {
+		name:    "handleEstimateStakeDiff: HeaderByHeight error",
+		handler: handleEstimateStakeDiff,
+		cmd:     &types.EstimateStakeDiffCmd{},
+		mockChain: func() *testRPCChain {
+			chain := defaultMockRPCChain()
+			chain.estimateNextStakeDifficultyQueue = validQueueFunc()
+			chain.headerByHeightErr = errors.New("")
+			return chain
+		}(),
+		wantErr: true,
+		errCode: dcrjson.ErrRPCInternal.Code,
+	}, {
+		name:    "handleEstimateStakeDiff: min diff estimation error",
+		handler: handleEstimateStakeDiff,
+		cmd:     &types.EstimateStakeDiffCmd{},
+		mockChain: func() *testRPCChain {
+			chain := defaultMockRPCChain()
+			queue := validQueueFunc()
+			queue[0].err = errors.New("")
+			chain.estimateNextStakeDifficultyQueue = queue
+			return chain
+		}(),
+		wantErr: true,
+		errCode: dcrjson.ErrRPCInternal.Code,
+	}, {
+		name:    "handleEstimateStakeDiff: high diff estimation error",
+		handler: handleEstimateStakeDiff,
+		cmd:     &types.EstimateStakeDiffCmd{},
+		mockChain: func() *testRPCChain {
+			chain := defaultMockRPCChain()
+			queue := validQueueFunc()
+			queue[1].err = errors.New("")
+			chain.estimateNextStakeDifficultyQueue = queue
+			return chain
+		}(),
+		wantErr: true,
+		errCode: dcrjson.ErrRPCInternal.Code,
+	}, {
+		name:    "handleEstimateStakeDiff: expected diff estimation error",
+		handler: handleEstimateStakeDiff,
+		cmd:     &types.EstimateStakeDiffCmd{},
+		mockChain: func() *testRPCChain {
+			chain := defaultMockRPCChain()
+			queue := validQueueFunc()
+			queue[2].err = errors.New("")
+			chain.estimateNextStakeDifficultyQueue = queue
+			return chain
+		}(),
+		wantErr: true,
+		errCode: dcrjson.ErrRPCInternal.Code,
+	}, {
+		name:    "handleEstimateStakeDiff: user diff estimation error",
+		handler: handleEstimateStakeDiff,
+		cmd: &types.EstimateStakeDiffCmd{
+			Tickets: dcrjson.Uint32(1),
+		},
+		mockChain: func() *testRPCChain {
+			chain := defaultMockRPCChain()
+			queue := validQueueFunc()
+			queue[3].err = errors.New("")
+			chain.estimateNextStakeDifficultyQueue = queue
+			return chain
+		}(),
+		wantErr: true,
+		errCode: dcrjson.ErrRPCInternal.Code,
+	}})
+}
+
```

Comment:
Test cases look good, hitting every case.  It's probably worth defining 
```
testChainParams.StakeDiffWindowSize = 144
```
in `defaultChainParams` so that if that value changes in `chaincfg.MainNetParams()` it won't potentially break these tests.

Reply:
Would be surprised if that changed, but good idea nonetheless.

Reply:
Agreed, this and some of the other defaultChainParams are not likely to change, but I think it makes sense for consistency as well.